File: Solutions/PE86.py
# Project Euler 86

M = 100
perfect_squares = set()
for a in xrange(1,int(2.24*M)):
    perfect_squares.add(a**2)

def f(M):
    number = 0
    for a in xrange(1,M+1):
        for b in xrange(1,a+1):
            for c in xrange(1,b+1):
                result = (b+c)**2+a**2
                if result in perfect_squares:
                    number += 1
    return number

File: Solutions/PE86.cpp
#include <iostream>
#include <math.h>
using namespace std;

int f(int M) {
  int number = 0;
  for (int a=1;a<=M;a++) {
    for (int b=1;b<=a;b++) {
      for (int c=1;c<=b;c++) {
	  double result = (b+c)*(b+c) + a*a;
	  double s = sqrt(result) ;
	    if (s == (int)(s)) {
	      number++;
	    }
	}
    }
  }
  return number;
}

int main() {
  cout << "The answer is " << f(1810) << endl;
  return 0;
}
